worlds/lingo/regions.py

```diff
@@ -39,6 +57,24 @@ def connect_entrance(regions: Dict[str, Region], source_region: Region, target_r
         if door.door not in player_logic.item_by_door.get(effective_room, {}):
             for region in player_logic.calculate_door_requirements(effective_room, door.door, world).rooms:
                 world.multiworld.register_indirect_condition(regions[region], connection)
+    
+    if not pilgrimage and world.options.enable_pilgrimage and is_acceptable_pilgrimage_entrance(entrance_type, world):
+        for part in range(1, 6):
+            pilgrimage_descriptor = f" (Pilgrimage Part {part})"
+            if source_region.name == "Menu":
+                pilgrim_source_region = source_region
```

Comment:
Wait what? Why? This has Menu connecting to for example "Orange Tower (Pilgrimage Part 4)", which seems strange to me

Reply:
Oh, nevermind, that's one of the "conceptual" regions innit

Reply:
I get it now. I wonder why these meta regions exists instead of just having the door belong to one of the real regions. Naming?

Reply:
You're right, it's not dead code, but it's code with no impact. I see what the question was now. The main reason that those meta regions exist is because of an implementation details where doors have an implicit requirement that you be able to access the region they're defined in for them to be considered openable in vanilla doors. The orange tower doors have weird requirements for opening them, and the sunwarp doors don't exist in the base game. It also makes it easier to reference the doors.